aws/resource_aws_s3_bucket.go

```diff
@@ -2777,13 +2784,76 @@ func flattenS3ObjectLockConfiguration(conf *s3.ObjectLockConfiguration) []interf
 	return []interface{}{mConf}
 }
 
-func flattenGrants(ap *s3.GetBucketAclOutput) []interface{} {
-	//if ACL grants contains bucket owner FULL_CONTROL only - it is default "private" acl
-	if len(ap.Grants) == 1 && aws.StringValue(ap.Grants[0].Grantee.ID) == aws.StringValue(ap.Owner.ID) &&
-		aws.StringValue(ap.Grants[0].Permission) == s3.PermissionFullControl {
-		return nil
+func flattenCannedACL(d *schema.ResourceData, ap *s3.GetBucketAclOutput) bool {
+	var owner_full_control bool
+	var all_users_read bool
+	var all_users_write bool
+	var authenticated_users_read bool
+	var ec2_read bool
+	var logdelivery_read_acp bool
+	var logdelivery_write bool
+	// There may be more than one occurrence of each grant combination but it is apparently unimportant as
+	// the AWS console states "The console displays combined access grants for duplicate grantees"
```

Comment:
Example of such a case:
```
# aws --profile myprofile s3api get-bucket-acl --bucket mybucket
{
    "Owner": {
        "DisplayName": "awsroot.myaccount",
        "ID": "98a033a4faed3******146099526853"
    },
    "Grants": [
        {
            "Grantee": {
                "DisplayName": "awsroot.myaccount",
                "ID": "98a033a4faed3******146099526853",
                "Type": "CanonicalUser"
            },
            "Permission": "FULL_CONTROL"
        },
        {
            "Grantee": {
                "DisplayName": "awsroot.myaccount",
                "ID": "98a033a4faed3******146099526853",
                "Type": "CanonicalUser"
            },
            "Permission": "FULL_CONTROL"
        }
    ]
}
```

Reply:
Also found a case where there are three duplicate grants on another bucket in the same account:

```
# aws --profile myprofile s3api get-bucket-acl --bucket myotherbucket
{
    "Owner": {
        "DisplayName": "awsroot.myaccount",
        "ID": "98a033a4faed389d1c9e409dab5aa9aede0b78c92084b551a946146099526853"
    },
    "Grants": [
        {
            "Grantee": {
                "DisplayName": "awsroot.myaccount",
                "ID": "98a033a4faed3******146099526853",
                "Type": "CanonicalUser"
            },
            "Permission": "FULL_CONTROL"
        },
        {
            "Grantee": {
                "DisplayName": "awsroot.myaccount",
                "ID": "98a033a4faed3******146099526853",
                "Type": "CanonicalUser"
            },
            "Permission": "FULL_CONTROL"
        },
        {
            "Grantee": {
                "DisplayName": "awsroot.myaccount",
                "ID": "98a033a4faed3******146099526853",
                "Type": "CanonicalUser"
            },
            "Permission": "FULL_CONTROL"
        }
    ]
}
```